frontend/src/pages/MyCourses/MyCoursesContainer.tsx

```diff
@@ -7,19 +7,42 @@ import MyCoursesList from './MyCoursesList';
 
 const MyCoursesContainer: React.FC = () => {
   const { data, isLoading } = useGetMyCourses();
-  const windowWidth =
+  const disableLinkWidth =
     window.innerWidth < WINDOW_SIZE.sm.width ? WINDOW_SIZE.xs.name : WINDOW_SIZE.sm.name;
   const disableLink = (event: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => {
-    if (windowWidth === WINDOW_SIZE.sm.name) {
+    if (disableLinkWidth === WINDOW_SIZE.sm.name) {
       event.preventDefault();
     }
   };
 
+  const [isFilterOpen, setFilterOpen] = useState<boolean>(false);
+  const handleFilterOpen = () => {
+    setFilterOpen(!isFilterOpen);
+  };
+  const handleFilterClose = () => {
+    setFilterOpen(false);
+  };
+
+  let windowWidth = WINDOW_SIZE.xl.name;
+  if (window.innerWidth < WINDOW_SIZE.xl.width && window.innerWidth > WINDOW_SIZE.lg.width) {
+    windowWidth = WINDOW_SIZE.lg.name;
+  }
+  if (window.innerWidth < WINDOW_SIZE.lg.width && window.innerWidth > WINDOW_SIZE.md.width) {
+    windowWidth = WINDOW_SIZE.md.name;
+  }
+  if (window.innerWidth < WINDOW_SIZE.md.width && window.innerWidth > WINDOW_SIZE.sm.width) {
+    windowWidth = WINDOW_SIZE.sm.name;
+  }
```

Comment:
Do we really need to duplicate those conditions in 3 components?

Reply:
Removed